### PR TITLE
[perf-test] update 100 RayJob perf tests to use PyTorch trainer and Ray Data examples

### PR DESCRIPTION
**Pull request by andrewsykim**
## Why are these changes needed?

Updates the 100 RayJob perf test to use pytorch trainer and Ray Data example workloads

## Related issue number

[](https://github.com/ray-project/kuberay/issues/2069)

## Checks

- [ ] I've made sure the tests are passing.
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by andrewsykim** [comment]
cc @kevin85421

File: benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+              image: {{.Image}}
+              ports:
+                - containerPort: 6379
+                  name: gcs-server
+                - containerPort: 8265
+                  name: dashboard
+                - containerPort: 10001
+                  name: client
+              env:
+              - name: "OMP_NUM_THREADS"
```
Comment (kevin85421) [inline benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml:31]: Move to `runtimeEnvYAML` instead? Could you also add some comments in the YAML to explain why do we need to set this? Thanks!

File: benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        maxReplicas: 5
+        groupName: worker-group
+        rayStartParams: {}
+        template:
+          spec:
+            containers:
+              - name: ray-worker
+                image: {{.Image}}
+                env:
+                - name: "OMP_NUM_THREADS"
```
Comment (kevin85421) [inline benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml:51]: Move to `runtimeEnvYAML` instead?

File: benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml (hunk 1 above)
Comment (andrewsykim) [reply benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml]: done

File: benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml (hunk 2 above)
Comment (andrewsykim) [reply benchmark/perf-tests/100-rayjob/pytorch-mnist-rayjob.yaml]: done

**Review by kevin85421 — APPROVED** [review]
Thanks

**Pull request merged**